### PR TITLE
ADIOS2: Add all dependencies as library products

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: A/ADIOS2/build_tarballs.jl [hunk 1]
```diff
@@ -78,8 +78,15 @@ platforms = expand_cxxstring_abis(platforms; skip=Sys.isapple)
...
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("libadios2_c_mpi", :libadios2_c_mpi),
+    LibraryProduct("libadios2_atl", :libadios2_atl),
     LibraryProduct("libadios2_c", :libadios2_c),
+    LibraryProduct("libadios2_c_mpi", :libadios2_c_mpi),
+    LibraryProduct("libadios2_core", :libadios2_core),
+    LibraryProduct("libadios2_core_mpi", :libadios2_core_mpi),
+    LibraryProduct("libadios2_dill", :libadios2_dill),
+    LibraryProduct("libadios2_evpath", :libadios2_evpath),
```
Comment (giordano) [inline A/ADIOS2/build_tarballs.jl:87]: The libraries don't seem to be compiled at all for Windows, as far as I could tell

Comment (eschnett) [reply A/ADIOS2/build_tarballs.jl:87]: Is there a way to have a different set of products for Windows?

Comment (giordano) [reply A/ADIOS2/build_tarballs.jl:87]: There is a way, but it isn't super user-friendly:
https://github.com/JuliaPackaging/Yggdrasil/blob/8be19820afa24ff817185f3c53c4d20786a8add3/M/MKL/build_tarballs.jl#L50-L65
The gist is that you need to call `build_tarballs()` multiple times, so you can have different arguments, for example different platforms (once all non Windows, `!Sys.iswindows`, the other for Windows, `Sys.iswindows`) and products.  We have ideas about how to make this less artisanal, but it isn't high-priority as there is already a work-around without changing anything in `BinaryBuilder`, as bad as it is.

However, if you don't plan to use these two libraries nor to add other packages depending on this one which may need these libraries, I'd suggest to simply leave them out.  But it's up to you

Comment (eschnett) [reply A/ADIOS2/build_tarballs.jl]: I'll try. I think `openPMD_api` might depend on some of these on non-Windows platforms.

I realize that I can do almost all I want by calling ADIOS2 directly, and I might not need openPMD_api.

**Comment by eschnett** [comment]
Ping?

**Pull request merged**